### PR TITLE
Deprecating TASK.SUCCESS_DISTANCE since TASK.SUCESS.SUCCESS_DISTANCE replaces it

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

In our config files, there are two separate SUCCESS_DISTANCE keys, one under TASK.SUCCESS_DISTANCE and one under TASK.SUCCESS.SUCESS_DISTANCE. It seems only the TASK.SUCCESS.SUCESS_DISTANCE is used since SUCESS is a measure on its own.

## How Has This Been Tested

Removed the TASK.SUCCESS_DISTANCE key and ran a few configs and the unit tests.

## Types of changes

- Docs change / refactoring / dependency upgrade

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by Skylion007** [comment]
Shouldn't we be using YACS deprecated keys? Or throwing an error if the key is found? Loading old configs could silently break otherwise.

**Comment by vincentpierre** [comment]
> Shouldn't we be using YACS deprecated keys? Or throwing an error if the key is found? Loading old configs could silently break otherwise.

You are right, I should do this. I find there is some trickiness here because of [this line](https://github.com/rbgirshick/yacs/blob/master/yacs/config.py#L484). YACS will only check if a new key is deprecated if adding new keys is not allowed. (I think it is a bug.) One option would be to check ourselves for deprecated keys (which will increase code complexity) or add to YACS deprecated keys and not have meaningful warnings when the user uses a deprecated key.

File: configs/test/habitat_mp3d_object_nav_test.yaml [hunk 1]
```diff
@@ -30,7 +30,6 @@ SIMULATOR:
 TASK:
   TYPE: ObjectNav-v1
   POSSIBLE_ACTIONS: ["STOP", "MOVE_FORWARD", "TURN_LEFT", "TURN_RIGHT", "LOOK_UP", "LOOK_DOWN"]
-  SUCCESS_DISTANCE: 0.1
```
Comment (mathfac) [inline configs/test/habitat_mp3d_object_nav_test.yaml:33]: Should we do `SUCCESS:
    SUCCESS_DISTANCE: 0.1` ?

**Review by mathfac — APPROVED** [review]
Thank you for addressing this!

**Pull request merged**